routers/web/repo/compare.go

```diff
@@ -917,25 +924,88 @@ func ExcerptBlob(ctx *context.Context) {
 			Type:    gitdiff.DiffLineSection,
 			Content: lineText,
 			SectionInfo: &gitdiff.DiffLineSectionInfo{
-				Path:          filePath,
-				LastLeftIdx:   lastLeft,
-				LastRightIdx:  lastRight,
-				LeftIdx:       idxLeft,
-				RightIdx:      idxRight,
-				LeftHunkSize:  leftHunkSize,
-				RightHunkSize: rightHunkSize,
+				Path:                filePath,
+				LastLeftIdx:         lastLeft,
+				LastRightIdx:        lastRight,
+				LeftIdx:             idxLeft,
+				RightIdx:            idxRight,
+				LeftHunkSize:        leftHunkSize,
+				RightHunkSize:       rightHunkSize,
+				HasComments:         false,
+				LastRightCommentIdx: 0,
+				RightCommentIdx:     0,
 			},
+			Comments: nil,
 		}
 		if direction == "up" {
 			section.Lines = append([]*gitdiff.DiffLine{lineSection}, section.Lines...)
 		} else if direction == "down" {
 			section.Lines = append(section.Lines, lineSection)
 		}
 	}
+	issueIndex := ctx.FormInt64("issue_index")
+	if ctx.FormBool("pull") && issueIndex > 0 {
+		issue, err := issues_model.GetIssueByIndex(ctx, ctx.Repo.Repository.ID, issueIndex)
+		if issue == nil {
+			ctx.ServerError("GetIssueByIndex", err)
+			return
+		}
+		allComments, err := issues_model.FetchCodeComments(ctx, issue, ctx.Doer, false)
```

Comment:
So we should just get the comments from the `filePath` but not load all the comments from the issue.